x/attribute/keeper/keeper.go

```diff
@@ -438,18 +449,26 @@ func (k Keeper) PurgeAttribute(ctx sdk.Context, name string, owner sdk.AccAddres
 	if err != nil {
 		return err
 	}
+	store := ctx.KVStore(k.storeKey)
 	for _, acct := range accts {
-		store := ctx.KVStore(k.storeKey)
-		it := sdk.KVStorePrefixIterator(store, types.AddrAttributesNameKeyPrefix(acct, name))
-		for ; it.Valid(); it.Next() {
-			store.Delete(it.Key())
+		attrToDelete := k.getAccountsToDelete(store, acct, name)
+		for _, key := range attrToDelete {
+			store.Delete(key)
 			k.DecAttrNameAddressLookup(ctx, name, acct)
 		}
-		it.Close()
 	}
 	return nil
 }
 
+func (k Keeper) getAccountsToDelete(store sdk.KVStore, acctAddr sdk.AccAddress, attributeName string) (attrToDelete [][]byte) {
```

Comment:
Probably want this called `getAttrsToDelete`

Reply:
Ooops, I was actually going to refactor that but forgot.  I named it to `getAccountAttributesByName` to be more specific to the actual method.  `getAttrsToDelete`  is specific to an expected caller... This method might be useful for something else in the future.